<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>895</width>
    <height>682</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>850</width>
      <height>601</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab_1">
     <attribute name="title">
      <string>Train model</string>
     </attribute>
     <widget class="QPushButton" name="modelLoader">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>30</y>
        <width>261</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>Load model configuration from file</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>70</y>
        <width>701</width>
        <height>391</height>
       </rect>
      </property>
      <property name="title">
       <string>Settings for training a model</string>
      </property>
      <widget class="QWidget" name="gridLayoutWidget">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>50</y>
         <width>481</width>
         <height>41</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="2">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>4</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignHCenter|Qt::AlignTop</set>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>3</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignHCenter|Qt::AlignTop</set>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>5</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignHCenter|Qt::AlignTop</set>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>2</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignHCenter|Qt::AlignTop</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QSlider" name="hiddenLayersAmount">
       <property name="geometry">
        <rect>
         <x>240</x>
         <y>30</y>
         <width>381</width>
         <height>23</height>
        </rect>
       </property>
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="maximum">
        <number>5</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="tracking">
        <bool>true</bool>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="invertedAppearance">
        <bool>false</bool>
       </property>
       <property name="invertedControls">
        <bool>false</bool>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBelow</enum>
       </property>
       <property name="tickInterval">
        <number>1</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>30</y>
         <width>221</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Hidden layers amount</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>90</y>
         <width>141</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Hidden layers sizes</string>
       </property>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget_2">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>90</y>
         <width>511</width>
         <height>58</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="3">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>layer 4</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>layer 3</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>layer 2</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>layer 1</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>layer 5</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QSpinBox" name="layerSize_1">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="layerSize_2">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSpinBox" name="layerSize_3">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QSpinBox" name="layerSize_4">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QSpinBox" name="layerSize_5">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QLabel" name="trainingDatasetDirectory">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>260</y>
         <width>361</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>/</string>
       </property>
      </widget>
      <widget class="QLabel" name="testingDatasetDirectory">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>290</y>
         <width>341</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>/</string>
       </property>
      </widget>
      <widget class="QPushButton" name="chooseTestingDataset">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>290</y>
         <width>171</width>
         <height>29</height>
        </rect>
       </property>
       <property name="text">
        <string>Choose testing dataset</string>
       </property>
      </widget>
      <widget class="QPushButton" name="chooseTrainingDataset">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>260</y>
         <width>171</width>
         <height>29</height>
        </rect>
       </property>
       <property name="text">
        <string>Choose training dataset</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="graphMode">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>170</y>
         <width>112</width>
         <height>26</height>
        </rect>
       </property>
       <property name="text">
        <string>graph</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="matrixMode">
       <property name="geometry">
        <rect>
         <x>230</x>
         <y>170</y>
         <width>112</width>
         <height>26</height>
        </rect>
       </property>
       <property name="text">
        <string>matrix</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>170</y>
         <width>91</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Model type</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_16">
       <property name="geometry">
        <rect>
         <x>230</x>
         <y>220</y>
         <width>101</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Batch size</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_17">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>220</y>
         <width>101</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Epoch amount</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_18">
       <property name="geometry">
        <rect>
         <x>410</x>
         <y>220</y>
         <width>121</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Learning constant</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="learningConstant">
       <property name="geometry">
        <rect>
         <x>560</x>
         <y>220</y>
         <width>67</width>
         <height>29</height>
        </rect>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
      <widget class="QSpinBox" name="epochAmount">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>220</y>
         <width>48</width>
         <height>29</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="batchSize">
       <property name="geometry">
        <rect>
         <x>320</x>
         <y>220</y>
         <width>48</width>
         <height>29</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>9999</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
      </widget>
      <widget class="QCheckBox" name="evaluateCheckBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>330</y>
         <width>341</width>
         <height>26</height>
        </rect>
       </property>
       <property name="text">
        <string>Evaluate model performance after each epoch</string>
       </property>
       <property name="tristate">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>360</y>
         <width>411</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>checking that box will significantly slow down performance</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="trainModel">
      <property name="geometry">
       <rect>
        <x>580</x>
        <y>510</y>
        <width>131</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>TRAIN MODEL</string>
      </property>
     </widget>
     <widget class="QPushButton" name="saveModel">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>510</y>
        <width>231</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>Save model to file</string>
      </property>
     </widget>
     <widget class="QPushButton" name="runExperiment">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>470</y>
        <width>291</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>Run the experiment on the test sample</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="testSampleFraction">
      <property name="geometry">
       <rect>
        <x>690</x>
        <y>470</y>
        <width>67</width>
        <height>29</height>
       </rect>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_25">
      <property name="geometry">
       <rect>
        <x>520</x>
        <y>470</y>
        <width>141</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Test sample fraction:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Research">
      <property name="geometry">
       <rect>
        <x>630</x>
        <y>10</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Research</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="numresearch">
      <property name="geometry">
       <rect>
        <x>640</x>
        <y>50</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>1001</number>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <property name="palette">
      <palette>
       <active/>
       <inactive/>
       <disabled/>
      </palette>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <attribute name="title">
      <string>Classify image</string>
     </attribute>
     <widget class="Paint" name="paint" native="true">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>512</width>
        <height>512</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="classifyImage">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>60</y>
        <width>151</width>
        <height>101</height>
       </rect>
      </property>
      <property name="text">
       <string>Classify</string>
      </property>
     </widget>
     <widget class="QLabel" name="answer">
      <property name="geometry">
       <rect>
        <x>670</x>
        <y>180</y>
        <width>63</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>180</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Your letter is:</string>
      </property>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>10</y>
        <width>211</width>
        <height>41</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="clear">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="load_img">
         <property name="text">
          <string>Load Img</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget_3">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>260</y>
        <width>281</width>
        <height>211</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="3" column="0">
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>Sizes of layers:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLabel" name="precision">
         <property name="text">
          <string>empty</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="layersSizes">
         <property name="text">
          <string>empty</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QLabel" name="recall">
         <property name="text">
          <string>empty</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLabel" name="accuracy">
         <property name="text">
          <string>empty</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_23">
         <property name="text">
          <string>Recall:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_22">
         <property name="text">
          <string>Precision:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Current model configuration</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>Accuracy</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Layers count:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="layersCount">
         <property name="text">
          <string>empty</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_24">
         <property name="text">
          <string>F1-measure:</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QLabel" name="f1Measure">
         <property name="text">
          <string>empty</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="hiddenLayersCount_4">
    <property name="geometry">
     <rect>
      <x>980</x>
      <y>340</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>895</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Paint</class>
   <extends>QWidget</extends>
   <header location="global">paint.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
